rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -238,10 +238,13 @@ NodeGraph::wait_for_graph_change(
       throw EventNotRegisteredError();
     }
   }
+  auto pred = [&event]() {
+      return event->check() || !rclcpp::utilities::ok();
+    };
   std::unique_lock<std::mutex> graph_lock(graph_mutex_);
-  graph_cv_.wait_for(graph_lock, timeout, [&event]() {
-    return event->check() || !rclcpp::utilities::ok();
-  });
+  if (!pred()) {
+    graph_cv_.wait_for(graph_lock, timeout, pred);
+  }
```

Comment:
This reference says that, after a fashion, `wait_for` is equivalent to:

> Equivalent to
```c++
while (!pred()) {
    if (wait_until(lock, abs_time) == std::cv_status::timeout) {
        return pred();
    }
}
return true;
```
>  This overload may be used to ignore spurious wakeups.

-- http://en.cppreference.com/w/cpp/thread/condition_variable/wait_until

However, it does not appear to be guaranteed (or at least to my satisfaction) that the predicate is checked before entering to block. So this extra `if (!pred())` is really just defensive to changes in the std libraries implementation. On the other hand, it is a really cheap check and shouldn't do any harm.